### PR TITLE
Fix ExtractLevels button

### PR DESCRIPTION
**Pull request by mkeshavarzi**
Continuation of #356 to enable toggling of level cut planes and preserving hash params.

### CODE REVIEW

File: src/Components/ExtractLevelsMenu.jsx [hunk 1]
```diff
@@ -110,11 +125,12 @@ export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
           <MenuItem
             key={i}
             onClick={() =>
-              createFloorplanPlane(allStoreys[i] + floorOffset, allStoreys[i + 1] - ceilingOffset)}
+              // ADDED SELECTED:
+              createFloorplanPlane(allStoreys[i] + floorOffset, allStoreys[i + 1] - ceilingOffset)} selected={
```
Comment (OlegMoshkovich) [inline src/Components/ExtractLevelsMenu.jsx:129]: move selected to the next line.

File: src/Components/ExtractLevelsMenu.jsx [hunk 2]
```diff
@@ -41,6 +46,17 @@ export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
     const normal2 = new Vector3(0, -1, 0)
     viewer.clipper.createFromNormalAndCoplanarPoint(normal1, modelCenter1)
     viewer.clipper.createFromNormalAndCoplanarPoint(normal2, modelCenter2)
+    console.log(h1)
```
Comment (OlegMoshkovich) [inline src/Components/ExtractLevelsMenu.jsx:49]: remove. please proofread the code, and delete all of the consoles.

File: src/Components/ExtractLevelsMenu.jsx [hunk 3]
```diff
@@ -33,6 +33,11 @@ export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
     setAnchorEl(null)
   }
   const viewer = useStore((state) => state.viewerStore)
+  // ADDED
```
Comment (OlegMoshkovich) [inline src/Components/ExtractLevelsMenu.jsx:36]: remove the comments. 
for the review purposes leave the comments in the PR.

File: src/Components/ExtractLevelsMenu.jsx [hunk 4]
```diff
@@ -33,6 +33,11 @@ export default function ExtractLevelsMenu({listOfOptions, icon, title}) {
...
+  const levelInstance = useStore((state) => state.levelInstance)
+  const setLevelInstance = useStore((state) => state.setLevelInstance)
```
Comment (OlegMoshkovich) [inline src/Components/ExtractLevelsMenu.jsx:38]: did you add these variables to the store?
image
if not add it to the UI slice in the store folder.

**Comment by OlegMoshkovich** [comment]
image

for the name of the feature - I would use isolate levels, instead of extract levels

File: src/Components/ExtractLevelsMenu.jsx (hunk 4 above)
Comment (mkeshavarzi) [reply src/Components/ExtractLevelsMenu.jsx:37]: Addressed. Added to UI Slice

File: src/Components/ExtractLevelsMenu.jsx (hunk 3 above)
Comment (mkeshavarzi) [reply src/Components/ExtractLevelsMenu.jsx]: Addressed. Removed logs.

File: src/Components/ExtractLevelsMenu.jsx (hunk 2 above)
Comment (mkeshavarzi) [reply src/Components/ExtractLevelsMenu.jsx]: Addressed. Changed names.

File: src/Components/ExtractLevelsMenu.jsx (hunk 1 above)
Comment (mkeshavarzi) [reply src/Components/ExtractLevelsMenu.jsx]: Addressed. Thanks

**Comment by mkeshavarzi** [comment]
> for the name of the feature - I would use isolate levels, instead of extract levels

Changed name. Also addressed in tests.

**Comment by mkeshavarzi** [comment]
@OlegMoshkovich executing the "Isolate Levels" function with hash params for sharing the link seems to be working now. Would appreciate it if you take a look. Thanks!

**Comment by OlegMoshkovich** [comment]
great work on the levels in the url.
there is a method in the operationsGroup that is responsible for highlighting the clear button.
would be nice to add logic to highlight the button when the level is isolated.
image

**Comment by mkeshavarzi** [comment]
> great work on the levels in the url. there is a method in the operationsGroup that is responsible for highlighting the clear button. would be nice to add logic to highlight the button when the level is isolated. 

Done!

**Comment by mkeshavarzi** [comment]
> great work on the levels in the url. there is a method in the operationsGroup that is responsible for highlighting the clear button. would be nice to add logic to highlight the button when the level is isolated.

@OlegMoshkovich - Done :)

**Comment by OlegMoshkovich** [comment]
Almost done!
For some reason the clear button is highlighted when a level is selected and stays highlighted even when the level is cleared.

Something is not reset correctly.
Please fix.
Let me know if you don't get it within an hour.

**Comment by mkeshavarzi** [comment]
> Almost done! For some reason the clear button is highlighted when a level is selected and stays highlighted even when the level is cleared.
> 
> Something is not reset correctly. Please fix. Let me know if you don't get it within an hour.

Great catch. This also happens in CutPlanes. I'll take a look.

**Comment by OlegMoshkovich** [comment]
@mkeshavarzi I fixed it on your branch.
image

**Pull request merged**